src/Options/OptionsDetails.hpp

```diff
@@ -173,12 +174,15 @@ struct print;
 template <typename T, typename OptionList, typename = std::nullptr_t>
 struct print_impl {
   // T is a group here
```

Comment:
I'm unsure if the compiler will complain about this, but it might be good to format this as:

```cpp
template <typename GroupOrTag, typename OptionList, typename = std::nullptr_t>
struct print_impl;

template <typename Group, typename OptionList>
struct print_impl<Group, OptionList, std::nullptr_t> {
  ...
};

template <typename Tag, typename OptionList>
struct print_impl<.....
```

If that works then I think the code will be easier to understand.

Reply:
That errors with `ambiguous partial specializations`, so I had to omit the forward declaration